azure/converters/spotinstances_test.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package converters
+
+import (
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2021-04-01/compute"
+	"github.com/Azure/go-autorest/autorest/to"
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/api/resource"
+	infrav1 "sigs.k8s.io/cluster-api-provider-azure/api/v1beta1"
+)
+
+func Test_GetSpotVMOptions(t *testing.T) {
+	type resultParams struct {
+		vmPriorityTypes       compute.VirtualMachinePriorityTypes
+		vmEvictionPolicyTypes compute.VirtualMachineEvictionPolicyTypes
+		billingProfile        *compute.BillingProfile
+	}
+	cases := []struct {
+		name          string
+		spotVMOptions *infrav1.SpotVMOptions
+		expectErr     bool
+		expect        func(*GomegaWithT, resultParams)
+	}{
+		{
+			name:          "should return empty vm priority, empty eviction policy and nil billing profile for nil spotVMOptions",
+			spotVMOptions: nil,
+			expectErr:     false,
```

Comment:
Can we use `expectErr` as a string like in image_test.go and add a test case where we are returning an error?